src/frank.c

```diff
@@ -81,42 +78,43 @@ SEXP frank(SEXP xorderArg, SEXP xstartArg, SEXP xlenArg, SEXP ties_method) {
   else if (!strcmp(CHAR(STRING_ELT(ties_method, 0)), "sequence")) ties = SEQUENCE;
   // else if (!strcmp(CHAR(STRING_ELT(ties_method, 0)), "runlength")) ties = RUNLENGTH;
   else error(_("Internal error: invalid ties.method for frankv(), should have been caught before. please report to data.table issue tracker")); // # nocov
-  n = length(xorderArg);
-  SEXP ans = (ties == MEAN) ? PROTECT(allocVector(REALSXP, n)) : PROTECT(allocVector(INTSXP, n));
-  int *ians = INTEGER(ans);
-  double *dans = REAL(ans);
-  if (n > 0) {
+  const int n = length(xorderArg);
+  SEXP ans = PROTECT(allocVector(ties==MEAN ? REALSXP : INTSXP, n));
+  int *ians=NULL;
```

Comment:
why initialize to `NULL`, can't we just do `int *ians;` and `double *dans;`?

Reply:
AFAIU to avoid compiler warnings about uninitialized variables